ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/QuantityWrapper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.core;
+
+import org.hyperledger.besu.plugin.data.Quantity;
+
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class QuantityWrapper implements Quantity {
```

Comment:
I'm not liking the wrappers for UInt256s.  The plugin API design was done in such a way that wrapping was not supposed to be done and we would pass the same objects to the plugin APIs just with restricted APIs.  We will need extensible UInt256s and bytes to get what we need.  This amount of wrapping will not be good for performance.

Reply:
This is going to require some rethinking on my end. I can look into that.

Reply:
Perhaps make all the methods and fields final or private/package private but not the class?  That should preserve the integrity of the superclass.  Then replace the wrappers with `DataBytes` and `UInt256Quantity` which extend the delegating classes and implement the plugins apis.  Both should only need constructors.

Reply:
I would plead that we look at this after this PR is merged. The wrappers are used seldomly I think, though I will easily grant having two methods `internalGetExtraData` vs `getExtraData` is really disgusting.